auto_scan.py

```diff
@@ -369,9 +382,22 @@ def deduplicate_scans(lw_client, start_time, end_time, container_scan_queue, reg
         # Skip if the container has a previously failed scan
         if qualified_repo in failed_scan_cache.keys():
             if image_id in failed_scan_cache[qualified_repo].keys():
-                logger.info(f'Skipping previously failed {qualified_repo} with image ID "{image_id}"')
-                skipped_containers += 1
-                continue
+
+                current_time = int(time.time())
+                try:
+                    if current_time > failed_scan_cache[qualified_repo][image_id]['expiry']:
+                        logger.info(f'Removing previously failed scan result for {qualified_repo} with image ID '
+                                    f'"{image_id}" due to expiration')
+                        del failed_scan_cache[qualified_repo][image_id]
+                    else:
+                        logger.info(f'Skipping previously failed {qualified_repo} with image ID "{image_id}"')
+                        skipped_containers += 1
+                        continue
+                except KeyError as err:
```

Comment:
The only thought that this approach raises on my end is that an unaccounted-for edge case could result in the cache being purged constantly and rendering the caching useless. We do have logging when this happens, so I think it's fine until/unless it becomes an issue down the road.